iceoryx_posh/test/integrationtests/test_publisher_subscriber_communication.cpp

```diff
@@ -126,6 +157,111 @@ TEST_F(PublisherSubscriberCommunication_test, AllSubscriberInterfacesCanBeSubscr
     }
 }
 
+TEST_F(PublisherSubscriberCommunication_test, SubscriberRequiringHistoryDoesNotConnectToPublisherWithoutHistorySupport)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "31cbd36d-32f1-4bc7-9980-0cdf5f248035");
+
+    constexpr uint64_t historyRequest = 1;
+    constexpr uint64_t historyCapacity = 0;
+    constexpr bool requireHistory = true;
+
+    auto publisher = createPublisher<int>(historyCapacity);
+    auto subscriber = createSubscriber<int>(historyRequest, requireHistory);
+
+    this->InterOpWait();
+
+    ASSERT_TRUE(publisher);
```

Comment:
Could we send also a simple integer from the publisher to the subscriber to verify that `subscriber` is subscribed and not something else.
It is a communication test but we do not perform any kind of actual  communication here.

I would add something for all the tests.
When they do not connect then sending an int should run into nothing otherwise the subscriber should receive something.

Reply:
Well, in this test it must not be subscribed

Reply:
We can do so, but actually I consider this wasteful since the tests run in isolation. We should separate tests which check matching and actual data transfer to reduce redundancy of tests. We currently do not have proper connection matching/discovery checks for many cases I think.

I would prefer the rationale that we

1. check whether we establish connections in all cases we need to
2. test whether we can transmit data whenever a connection is established but only for one connected case (otherwise this will scale badly if we add more cases where we expect connections)